prybar_assets/nodejs/input-sync.js

```diff
@@ -142,115 +123,22 @@ function checkForSigs(char) {
   }
 }
 
-/**
- * Writes a string at an index of another string, appending as needed.
- *
- * @param {string} str The base string
- * @param {string} other The new string which is being written
- * @param {number} index The index at which the new string should start at.
- * @return {string} str with other written at index.
- */
-function insertAt(str, other, index) {
-  return [str.slice(0, index) + other + str.slice(index), index + other.length];
-}
-
-/**
- * The ANSI escape code used to clear the contents of the current line to the right
- * of the cursor.
- */
-const escapeClearLineRight = "\x1b[K";
-
-/**
- * The escape used to move the cursor to a specific position in-line.
- *
- * @param {number} columnNum The position (starting at 1) in the current line which the cursor should be moved to.
- *
- */
-function escapeMoveCursorToColumn(columnNum) {
-  return `\x1b[${columnNum}G`;
-}
-
-/**
- * Sets the current line to our promt + string w/ the cursor at the right index.
- *
- * @param {string} prompt The question's prompt
- * @param {string} current The current string (what the user has input so far)
- * @param {number} index The index that
- */
-function displayPromptAndStr(prompt, current, index) {
-  writeTTYOutput(
-    // reset cursor position
-    "\r" +
-      // clear the rest of the line
-      // EL (Erase in Line ): in this case, as no number is speciifed,
-      // erases everything to the right of the cursor.
-      escapeClearLineRight +
-      // write the prompt
-      prompt +
-      // write the string
-      current +
-      escapeMoveCursorToColumn(prompt.length + index + 1)
-  );
-}
-
 /**
  * Synchronously reads from stdin until `\n` or `\r`
  *
  * @param {string} prompt The prompt to be displayed
  * @return {string} The input read (excluding newlines)
  */
 function question(prompt) {
-  return ensureRawMode(() => {
-    let str = "";
-    let index = 0;
+  let result = null;
 
-    if (!isTTY) {
-      writeOutput(prompt);
-    }
-
-    for (;;) {
-      displayPromptAndStr(prompt, str, index);
-      const didRead = readByteSync();
-
-      if (!didRead) {
-        return str;
-      }
-
-      const char = buf.toString("binary");
-      checkForSigs(char);
-
-      if (char === "\n" || char === "\r") {
-        writeTTYOutput("\r\n");
-
-        return str;
-      } else if (isTTY && char === "\x1b") {
-        const ret = handleArrowKey();
+  rl.question(prompt, (d) => {
+    result = d;
+  });
 
-        // if ret is a number, its the difference for the index
-        if (typeof ret === "number") {
-          // Only move the cursor if it will be in a valid position.
-          const newIndex = index + ret;
-          // the index can be equal to the strs length, if that's the case we're appending to the string.
-          if (newIndex >= 0 && newIndex <= str.length) {
-            index = newIndex;
-          }
+  while (result == null) rd.readNext();
```

Comment:
Is this `eqeq` intentional? I don't [see any case](https://nodejs.org/api/readline.html#rlquestionquery-options-callback_1) that the coercion would be desirable.

If it is intentional, maybe leave a comment explaining it.

Reply:
It is intentional.  It's what's called a "nullish" check.  it's somewhat unstandard but handy to know. 

`a == null` returns true if a is nullish (`null` or `undefined`).